js/plugins/api.js

```diff
@@ -1,25 +1,173 @@
+         * @return {String}         The full URL
+         */
+        build(url, params) {
+            if (!this.isAbsolute(url)) {
+                url  = `${this.ROOT}/${url}`;
+            }
+            if (params) {
+                const parts = [url, this.encodeParams(params)];
+                url = url.indexOf('?') >= 0 ? parts.join('&') : parts.join('?');
+            }
+            return url;
+        }
+
+        /**
+         * Default handler for JSON response
+         * @param  {Response} response A response with JSON expected body.
+         * @return {Promise}           A promise for data handling
+         */
+        jsonHandler(response) {
+            // TODO: error handling
+            return response.json();
+        }
+
+        /**
+         * Perform a GET operation and process JSON
+         * @param  {String} url     An absolute or API root relative URL
+         * @param  {Object} params  Optionnal query strings parameters
+         * @param  {[type]} headers Optionnal HTTP headers
+         * @return {Promise}        An instanciated fetch promise
+         */
+        get(url, params, headers) {
+            return this.getRaw(url, params, headers).then(this.jsonHandler);
+        }
+
+        /**
+         * Perform a GET operation
+         * @param  {String} url     An absolute or API root relative URL
+         * @param  {Object} params  Optionnal query strings parameters
+         * @param  {[type]} headers Optionnal HTTP headers
+         * @return {Promise}        An instanciated fetch promise
+         */
+        getRaw(url, params, headers) {
+            return fetch(this.build(url, params), {
+                method: 'get',
+                credentials: 'include',
+                headers: Object.assign({}, DEFAULT_HEADERS, headers || {})
+            });
+        }
+
+        /**
+         * Perform a POST operation with a JSON payload
+         * @param  {String} url     An absolute or API root relative URL
+         * @param  {Object} data    The body payload
+         * @param  {[type]} headers Optionnal HTTP headers
+         * @return {Promise}        An instanciated fetch promise
+         */
+        post(url, data, headers) {
+            return fetch(this.build(url), {
+                method: 'post',
+                credentials: 'include',
+                headers: Object.assign({}, DEFAULT_HEADERS, WRITE_HEADERS, headers || {}),
+                body: JSON.stringify(DATA)
+            }).then(this.jsonHandler);
+        }
+
+        /**
+         * Perform a POST upload operation (form/multipart)
+         * @param  {String} url     An absolute or API root relative URL
+         * @param  {Object} data    The body payload
+         * @param  {Object} headers Optionnal HTTP headers
+         * @return {Promise}        An instanciated fetch promise
+         */
+        upload(url, data, headers) {
+            const body = data;
```

Comment:
API use JSON everywhere, upload is a special case using `multipart/form-data` content type.
I just don't have implented the body yet.